File: Tarea7/src/Banco.java
import java.util.ArrayList;
import java.util.Scanner;
import java.util.Random;

class Banco {
    ArrayList<Employee> employeeList = new ArrayList<>();

    Scanner sc = new Scanner(System.in);
    Random num = new Random();

    public void ingresarDatos() {
        System.out.println("-Nombre:");
        String name = sc.next();
        System.out.println("-Apellido:");
        String lastName = sc.next();
        Employee employee = new Employee(name, lastName);
        employeeList.add(employee);
        System.out.println("Se han registrado con éxito los datos");
        System.out.println("****************");
    }

    public void mostrarEmpleados() {
        if (employeeList.isEmpty()) {
            System.out.println("*****  V  A  C  I  O  ******");
            System.out.println("Agregue un usuario con la opción '3'");
        } else {
            int i = 0;
            for (Employee employee : employeeList) {
                System.out.printf("Empleado %d:");
                System.out.printf("%s\n", i += 1, employee.getFullName());
            }
        }
    }

    public void mostrarInfoEmpleado() {
        if (employeeList.isEmpty()) {
            System.out.println("*****  V  A  C  I  O  ******");
            System.out.println("Agregue un usuario con la opción '3'");
        } else {
            System.out.println("¿De cuál empleado te gustaría saber su información?:");
            int i = sc.nextInt();
            if (i > employeeList.size()) {
                System.out.println("No se ha encontrado ningún empleado con ese número");
            } else {
                Employee employee = employeeList.get(i - 1);
                System.out.printf("Empleado: ");
                System.out.printf("%s\n", employee.getFullName());
                for (BankAccount account : employee.getAccountList()) {
                    System.out.println(account.getDatos());
                }
            }
        }
    }

    public void depositarDinero() {
        if (employeeList.isEmpty()) {
            System.out.println("*****  V  A  C  I  O  ******");
            System.out.println("No hay cuenta a la que depositar");
        } else {
            System.out.println("¿A qué empleado te gustaría depositar?:");
            int i = sc.nextInt();
            if (i > employeeList.size()) {
                System.out.println("No se ha encontrado el empleado");
            } else {
                Employee employee = employeeList.get(i - 1);
                System.out.println("¿A qué cuenta te gustaría depositar?:");
                int j = sc.nextInt();
                if (j > employee.getAccountList().size()) {
                    System.out.println("No se ha encontrado la cuenta");
                } else {
                    BankAccount account = employee.getAccountList().get(j - 1);
                    System.out.println("¿Cuánto desea depositar?");
                    double amount = sc.nextDouble();
                    account.agregarDinero(amount);
                    System.out.println("Se realizó el depósito con éxito.");
                }
            }
        }
    }

    public void agregarCuentaEmpleado() {
        if (employeeList.isEmpty()) {
            System.out.println("*****  V  A  C  I  O  ******");
            System.out.println("No hay empleado al que agregar una cuenta");
        } else {
            System.out.println("¿A qué empleado te gustaría agregar una cuenta?:");
            int i = sc.nextInt();
            if (i > employeeList.size()) {
                System.out.println("No se ha encontrado al empleado");
            } else {
                long accountNumber = num.nextInt(100000, 999999);
                System.out.println("Tipo de cuenta que desea agregar(A, B o C): ");
                char type = sc.next().charAt(0);
                BankAccount account = new BankAccount(accountNumber, type);
                Employee employee = employeeList.get(i - 1);
                employee.addAccount(account);
                System.out.println("Se ha registrado con éxito la cuenta");
                System.out.println("****************");
            }
        }
    }

    public void retirarDinero() {
        if (employeeList.isEmpty()) {
            System.out.println("*****  V  A  C  I  O  ******");
            System.out.println("No hay cuenta de la que retirar dinero");
        } else {
            System.out.println("¿De qué empleado te gustaría retirar dinero?:");
            int i = sc.nextInt();
            if (i > employeeList.size()) {
                System.out.println("No se ha encontrado el empleado");
            } else {
                Employee employee = employeeList.get(i - 1);
                System.out.println("¿De qué cuenta te gustaría retirar dinero?:");
                int j = sc.nextInt();
                if (j > employee.getAccountList().size()) {
                    System.out.println("No se ha encontrado la cuenta");
                } else {
                    BankAccount account = employee.getAccountList().get(j - 1);
                    System.out.println("¿Cuánto desea retirar?");
                    double amount = sc.nextDouble();
                    account.retirarDinero(amount);
                    System.out.println("Se realizó el retiro con éxito.");
                }
            }
        }
    }
}

File: Tarea7_python/Banco.py
import random
from Employee import Employee
from BanckAccount import BankAccount
class Banco:
  
  __employee_list = []
      

  def ingresar_datos(self):
      print("-Nombre:")
      name = input()
      print("-Apellido:")
      lastName = input()
      employee = Employee(name, lastName)
      self.__employee_list.append(employee)
      print("Se han registrado con éxito los datos")
      print("****************")

  def mostrar_empleados(self):
      if not self.__employee_list:
          print("*****  V  A  C  I  O  ******")
          print("Agregue un usuario con la opción '3'")
      else:
          i = 0
          for employee in self.__employee_list:
              print(f"Empleado {i + 1}: {employee.get_full_name()}")
              i += 1

  def mostrar_info_empleado(self):
      if not self.__employee_list:
          print("*****  V  A  C  I  O  ******")
          print("Agregue un usuario con la opción '3'")
      else:
          print("¿De cuál empleado te gustaría saber su información?:")
          i = int(input())
          if i > len(self.__employee_list):
              print("No se ha encontrado ningún empleado con ese número")
          else:
              employee = self.__employee_list[i - 1]
              print(f"Empleado: {employee.get_full_name()}")
              for account in employee.get_account_list():
                  print(account.get_datos())

  def depositar_dinero(self):
      if not self.__employee_list:
          print("*****  V  A  C  I  O  ******")
          print("No hay cuenta a la que depositar")
      else:
          print("¿A qué empleado te gustaría depositar?:")
          i = int(input())
          if i > len(self.__employee_list):
              print("No se ha encontrado el empleado")
          else:
              employee = self.__employee_list[i - 1]
              print("¿A qué cuenta te gustaría depositar?:")
              j = int(input())
              if j > len(employee.get_account_list()):
                  print("No se ha encontrado la cuenta")
              else:
                  account = employee.get_account_list()[j - 1]
                  print("¿Cuánto desea depositar?")
                  amount = float(input())
                  account.agregar_dinero(amount)
                  print("Se realizó el depósito con éxito.")

  def agregar_cuenta_empleado(self):
      if not self.__employee_list:
          print("*****  V  A  C  I  O  ******")
          print("No hay empleado al que agregar una cuenta")
      else:
          print("¿A qué empleado te gustaría agregar una cuenta?:")
          i = int(input())
          if i > len(self.__employee_list):
              print("No se ha encontrado al empleado")
          else:
              account_number = random.randint(100000, 999999)
              print("Tipo de cuenta que desea agregar (A, B o C): ")
              type = input()[0].upper()
              account = BankAccount(account_number, type)
              employee = self.__employee_list[i - 1]
              employee.add_account(account)
              print("Se ha registrado con éxito la cuenta")
              print("****************")

  def retirar_dinero(self):
      if not self.__employee_list:
          print("*****  V  A  C  I  O  ******")
          print("No hay cuenta de la que retirar dinero")
      else:
          print("¿De qué empleado te gustaría retirar dinero?:")
          i = int(input())
          if i > len(self.__employee_list):
              print("No se ha encontrado el empleado")
          else:
              employee = self.__employee_list[i - 1]
              print("¿De qué cuenta te gustaría retirar dinero?:")
              j = int(input())
              if j > len(employee.get_account_list()):
                  print("No se ha encontrado la cuenta")
              else:
                  account = employee.get_account_list()[j - 1]
                  print("¿Cuánto desea retirar?")
                  amount = float(input())
                  account.retirar_dinero(amount)
                  print("Se realizó el retiro con éxito.")